src/core/backend/pearl/core.rs

```diff
@@ -73,6 +75,40 @@ impl Pearl {
                 Error::failed(format!("cannot find actual alien folder. {:?}", operation))
             })
     }
+
+    async fn count_blobs(groups: Arc<Vec<Group>>, alien_groups_guard: Arc<RwLock<Vec<Group>>>,
+        count_interval: Duration) {
+        let mut interval = interval(count_interval);
+        loop {
+            interval.tick().await;
+
+            let mut cnt = 0;
+            for group in groups.iter() {
```

Comment:
`groups` is not protected by any synchronization primitive, so I think it is unsafe to iterate it in this manner

Reply:
why? for iterates over immutable reference.
Btw, @Justarone you can write:
```
for group in &groups {}
```

Reply:
Yes, there is no problem, because inside I call `group.holders()`, which returns cloned `Arc<RwLock<Vec<Holder>>>>` and then work with it (it's protected by sync primitivies)